standard/table.lua

```diff
@@ -0,0 +1,270 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Table
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+---
+-- @author Vogan for Liquipedia
+--
+-- A number of these functions take inspiration from Penlight: https://github.com/lunarmodules/Penlight
+--
+
+--local Class = require('Module:Class')
+
+local Table = {}
+
+function Table.randomize(tbl)
+	math.randomseed(os.time())
+
+	for i = #tbl, 2, -1 do
+		local j = math.random(i)
+		tbl[i], tbl[j] = tbl[j], tbl[i]
+	end
+	return tbl
+end
+
+function Table.size(tbl)
+	local i = 0
+	for _ in pairs(tbl) do
+		i = i + 1
+	end
+	return i
+end
+
+function Table.includes(tbl, value)
+	for _, entry in ipairs(tbl) do
+		if entry == value then
+			return true
+		end
+	end
+	return false
+end
+
+function Table.filter(tbl, predicate, argument)
+	local filteredTbl = {}
+	local foundMatches = 1
+
+	for _, entry in pairs(tbl) do
+		if predicate(entry, argument) then
```

Comment:
the predicate call here is `predicate(value)` which differs from most other functions in Table which calls `predicate(key, value)`. Should make them consistent eventually

Reply:
Nice one for a follow-up indeed